src/dbg/debugger.cpp

```diff
@@ -2116,39 +2202,66 @@ bool dbgsetcmdline(const char* cmd_line, cmdline_error_t* cmd_line_error)
     return true;
 }
 
-bool dbggetcmdline(char** cmd_line, cmdline_error_t* cmd_line_error)
+bool dbggetcmdline(char** cmd_line, cmdline_error_t* cmd_line_error, HANDLE hProcess /* = NULL */)
 {
     UNICODE_STRING CommandLine;
+    Memory<wchar_t*>* wstr_cmd;
     cmdline_error_t cmd_line_error_aux;
 
     if(!cmd_line_error)
         cmd_line_error = &cmd_line_error_aux;
 
-    if(!getcommandlineaddr(&cmd_line_error->addr, cmd_line_error))
+    if(!getcommandlineaddr(&cmd_line_error->addr, cmd_line_error, hProcess))
         return false;
 
-    if(!MemRead(cmd_line_error->addr, &CommandLine, sizeof(CommandLine)))
+    if(hProcess)
     {
-        cmd_line_error->type = CMDL_ERR_READ_PROCPARM_PTR;
-        return false;
-    }
+        duint NumberOfBytesRead;
+        if(!MemoryReadSafe(hProcess, (LPVOID)cmd_line_error->addr, &CommandLine, sizeof(UNICODE_STRING), &NumberOfBytesRead))
+        {
+            cmd_line_error->type = CMDL_ERR_READ_GETCOMMANDLINEBASE;
+            return false;
+        }
 
-    Memory<wchar_t*> wstr_cmd(CommandLine.Length + sizeof(wchar_t));
+        wstr_cmd = new Memory<wchar_t*>(CommandLine.Length + sizeof(wchar_t));

```

Comment:
This appears to be incorrect. I believe you should use a *C++* style declaration (eg not change this line at all and declare the variable only once its needed).